### PR TITLE
Implement QLinearAveragePool with unit tests.

### PR DESCRIPTION
**Pull request by zhanghuanrong**

### CODE REVIEW

File: onnxruntime/core/providers/cpu/nn/pool_base.h [hunk 1]
```diff
@@ -106,7 +106,8 @@ class PoolBase {
 
  protected:
   PoolBase(const OpKernelInfo& info)
-      : op_name_(info.GetKernelDef().OpName()),
+      : op_name_(info.GetKernelDef().OpName().rfind("QLinear", 0) != 0 ?
```
Comment (yufenglee) [inline onnxruntime/core/providers/cpu/nn/pool_base.h:109]: Will it have impact on other ops, like QLinearGlobalAveragePool? Is it better to handle it in the PoolAttributes

File: onnxruntime/contrib_ops/cpu/qlinear_pool.cc [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+      for (int64_t pw = 0; pw < pooled_width; ++pw) {
+        int64_t wstart = pw * stride_w - pads[1];
+        int64_t wend = std::min(wstart + kernel_shape[1], width);
+        wstart = std::max(wstart, static_cast<int64_t>(0));
+        const int64_t pool_index = ph * pooled_width + pw;
+        float Yh = PoolType::Initialize();
+        for (int64_t h = hstart; h < hend; ++h) {
+          for (int64_t w = wstart; w < wend; ++w) {
+            const int64_t input_index = h * width + w;
+            float x_value = dequantize_value(x_d[input_index], x_scale, x_zero_point);
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/qlinear_pool.cc:138]: Is it computed multiple time for different pooled_height? What if we dequantize the whole input at once, then AveragePool, and quantize it back, before we have an optimized version?

Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cpu/qlinear_pool.cc:138]: A good quick improve with some memory. Let me add some logic here.

File: onnxruntime/core/providers/cpu/nn/pool_base.h (hunk 1 above)
Comment (zhanghuanrong) [reply onnxruntime/core/providers/cpu/nn/pool_base.h:109]: Those are not affected. Also QLinear* is not using it as it is not supported here before.

**Pull request merged**